flycheck.el

```diff
@@ -7744,7 +7744,12 @@ Otherwise return the previously used cache directory."
 Look up the directory hierarchy from FILE for a directory
 containing a file that matches REGEXP."
   (locate-dominating-file
-   file
+   ;; We must not pass regular file here because
+   ;; `locate-dominating-file' will just pass it into its second
+   ;; lambda argument . But our lambda expects only directories.
+   (if (file-directory-p file)
+       file
+     (file-name-directory file))
```

Comment:
@cpitclaudel The issue here is that when passing `/usr/bin`, the search would start at `/usr` and not `/usr/bin`.   His argument is that can be surprising, since `/usr/bin` happens to be a directory.

So, it boils down to making a hard distinction between files and directories.